### ISSUE TITLE
Inconsistent results and false negative between Dev and CI

### ISSUE THREAD

**Issue by babysnakes** [fsprojects/FSharpLint#855]
Hi, I have a project using FSharpLint version  `0.26.10`  and Dotnet `10.0.203`. 

On my PC the linting passes without an error, but on CI I get this error:

```
========== Linting D:\a\GlazeWM-Tray\GlazeWM-Tray\src\TrayApp\TrayItem.fs ==========
Usage of `new` keyword here is redundant.
Error on line 49 starting at column 15
    let tray = new TrayIcon()
               ^             
See https://fsprojects.github.io/FSharpLint/how-tos/rules/FL0014.html
--------------------------------------------------------------------------------
```

The `TrayIcon` in the error, refers to Avalonia's `TrayIcon` which implements `IDisposable` so the `new` is there for a reason.

Both running on windows (I'm not sure which Windows version does Github Actions runs on).

Thanks

**Comment by Numpsy** [comment]
Is this running on a single file or a project?

Also are you restoring the project before linting it? I think I've seen an issue before with types in external libraries that haven't been restored, and I can't recal if the issue is still there.

**Comment by babysnakes** [comment]
Oops, sorry, you're right. The restore task runs only after the check 🙁.
